tests/rptest/tests/datalake/custom_partitioning_test.py

```diff
@@ -200,6 +205,9 @@ def test_basic(self, cloud_storage_type, catalog_type):
             # Check that files are correctly partitioned and that
             # spark can use this partitioning for a delete query.
 
+            table_name = f"redpanda.{topic_name}"
+            spark = dl.spark()
```

Comment:
should these tests cover Trino as well? I don't immediately smell an opportunity for divergence, but I haven't given it a ton of thought.

Reply:
That would be nice, but the queries that these tests use to verify partitioning are spark-dependent.

Reply:
fair enough. i can take an AI to look into building the Trino side as part of [CORE-9386](https://redpandadata.atlassian.net/browse/CORE-9386), assuming it's possible